Kudu.Services/ServiceHookHandlers/FetchHandler.cs

```diff
@@ -422,7 +423,11 @@ public static void PerformBackgroundDeployment(DeploymentInfo deployInfo, IEnvir
                 {"method", "POST"}
             });
 
-            Task.Run(() =>
+            // For monitoring creation of temp deployment
+            var tcs = new TaskCompletionSource<object>();
```

Comment:
TCS was a little "friendlier" for the situation. ARE/MRE require disposal, try/catch check for ObjectDisposedException when calling Set() (if the timeout fires first, we'll dispose it before Set() gets called), and you can await a TCS task directly.